test/unit/tasks/backend_api_test.rb

```diff
@@ -2,26 +2,41 @@
 
 module Tasks
   class BackendApiTest < ActiveSupport::TestCase
-    setup do
-      @backend_api = FactoryBot.create(:backend_api)
-      FactoryBot.create(:backend_api_config, backend_api: @backend_api)
-      @orphan_backend_api = FactoryBot.create(:backend_api)
-    end
+    class DestroyOrphans < ActiveSupport::TestCase
+      setup do
+        @backend_api = FactoryBot.create(:backend_api)
+        FactoryBot.create(:backend_api_config, backend_api: @backend_api)
+        @orphan_backend_api = FactoryBot.create(:backend_api)
+      end
+
+      test 'destroy orphans when account can not use api as product' do
+        Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(false).at_least_once
+        DeleteObjectHierarchyWorker.expects(:perform_later).with(@orphan_backend_api).once
+        DeleteObjectHierarchyWorker.expects(:perform_later).with(@backend_api).never
 
-    test 'destroy orphans when account can not use api as product' do
-      Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(false).at_least_once
-      DeleteObjectHierarchyWorker.expects(:perform_later).with(@orphan_backend_api).once
-      DeleteObjectHierarchyWorker.expects(:perform_later).with(@backend_api).never
+        execute_rake_task 'backend_api.rake', 'backend_api:destroy_orphans'
+      end
 
-      execute_rake_task 'backend_api.rake', 'backend_api:destroy_orphans'
+      test 'does not destroy orphans when account can use api as product' do
+        Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(true).at_least_once
+        DeleteObjectHierarchyWorker.expects(:perform_later).with(@orphan_backend_api).never
+        DeleteObjectHierarchyWorker.expects(:perform_later).with(@backend_api).never
+
+        execute_rake_task 'backend_api.rake', 'backend_api:destroy_orphans'
+      end
     end
 
-    test 'does not destroy orphans when account can use api as product' do
-      Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(true).at_least_once
-      DeleteObjectHierarchyWorker.expects(:perform_later).with(@orphan_backend_api).never
-      DeleteObjectHierarchyWorker.expects(:perform_later).with(@backend_api).never
+    test 'create_default_metrics' do
+      backend_apis_without_metrics = FactoryBot.create_list(:backend_api, 2)
+      backend_apis_without_metrics.each { |backend_api| backend_api.metrics.delete_all }
+      _backend_apis_with_metrics = FactoryBot.create_list(:backend_api, 2)
+
+      execute_rake_task 'backend_api.rake', 'backend_api:create_default_metrics'
 
-      execute_rake_task 'backend_api.rake', 'backend_api:destroy_orphans'
+      BackendApi.all.each do |backend_api|
+        assert_equal 1, backend_api.metrics.count
```

Comment:
Other tests can leak and raise false positives here. Better running only through the backend apis created in this test.

Reply:
Right, makes sense 👍